internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -418,12 +417,6 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		if err != nil {
 			return err
 		}
-
```

Comment:
We can remove the entire block that this is nested in i.e. from line 415-427 since without the locks these statements don't have any purpose.

Reply:
Hi @stephybun - These statements are still used in other places like https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_node_pool_resource.go#L674